interpreter/cling/lib/Interpreter/Exception.cpp

```diff
@@ -31,7 +31,7 @@ void cling_runtime_internal_throwIfInvalidPointer(void* Sema, void* Expr,
   else if (!cling::utils::isAddressValid(Arg))
     throw cling::InvalidDerefException(S, E,
           cling::InvalidDerefException::DerefType::INVALID_MEM);
-  return;
+  return const_cast<void*>(Arg);
```

Comment:
Can you add a comment why we take `const void*` and return `void*` (namely to reduce the complexity in the calling AST nodes, at the expense of possibly doing a `T*` -> `const void*` -> `const_cast<void*>` -> `T*` round trip)?